cpp/models/abm/location.cpp

```diff
@@ -37,7 +38,7 @@ Location::Location(LocationType type, uint32_t index, uint32_t num_cells)
     , m_capacity(LocationCapacity())
     , m_capacity_adapted_transmission_risk(false)
     , m_subpopulations(Eigen::Index(InfectionState::Count))
-    , m_cached_exposure_rate({AgeGroup::Count, VaccinationState::Count})
+    , m_cached_exposure_rate({AgeGroup(SimulationParameters::DEFAULT_NUM_AGE_GROUPS), VaccinationState::Count})
```

Comment:
This should somehow take into account the number of age groups known to the simulation/world. One could think about templating everything, similar to the CompartmentalModel.

Reply:
I add num_agegroups as a new parameter in the constructor so the user has to specify it when using Location.